src/headerssync.cpp

```diff
@@ -23,15 +27,34 @@ constexpr size_t REDOWNLOAD_BUFFER_SIZE{14827}; // 14827/624 = ~23.8 commitments
 // re-calculate parameters if we compress further)
 static_assert(sizeof(CompressedHeader) == 48);
 
+static size_t ComputeHeadersCacheMax(const CBlockIndex* chain_start, const std::optional<size_t> cache_bytes, size_t element_size)
+{
+    if (cache_bytes) {
+        return *cache_bytes / element_size;
+    }
+
+    return std::min<size_t>(
+        // 1.1 - Account for 10% more blocks than one every 10min.
+        1.1f * ((NodeClock::now() - NodeSeconds{std::chrono::seconds{chain_start->GetMedianTimePast()}}) / 10min),
```

Comment:
not sure it matters but we can't expect the other nets to have 10 blocks on average. Testnet having a block storm shouldn't crash the node (not sure it's the case, just commenting on assumptions)